src/view/status-first-page.cpp

```diff
@@ -354,35 +352,23 @@ static void display_first_page(PlayerType *player_ptr, int xthb, int *damage, in
     int xfos = player_ptr->skill_fos;
     int xdig = player_ptr->skill_dig;
 
-    concptr desc = likert(xthn, 12);
-    display_player_one_line(ENTRY_SKILL_FIGHT, desc, likert_color);
+    display_player_one_line(ENTRY_SKILL_FIGHT, likert(xthn, 12).data(), likert_color);
```

Comment:
likert_color is a file-scope static variable whose value is rewritten in the likert() function.
And since the order in which function arguments are evaluated is not specified, it is compiler-dependent whether likert_color has a value before or after the corresponding likert() call. This may cause the displayed color to be different than expected.
Since likert_color being a static variable can cause such confusing bugs, it is suggested that the return value of likert() be modified to return a pair of std::string and TERM_COLOR and that they be used.

Reply:
That explains the broken coloring on the first page of the character sheet.  The forced push eliminates the use of the file static variable.